src/controllers/admin/event.ts

```diff
@@ -52,52 +52,108 @@ export const GetDraftEventsHandler = async (_: Request, res: Response) => {
   }
 }
 
-export const handleGetReleasedEvents = async (_: Request, res: Response) => {
+export const CreateEventHandler = async (req: Request, res: Response) => {
+
+  const validevent = EventValidator.safeParse(req.body);
+
+  if(!validevent.success){
+    console.log(validevent.error);
+    return res.status(400).json({
+      message: "Invalid Event data provided!"
+    });
+  }
+
+  const {name,startTime,endTime,guests,venue,posterUrl,recordingURL,tags,status,entry,mode,eventFee} = req.body;
```

Comment:
This is unnecessary. validevent.data contains all the required fields. Please change accordingly.

Reply:
Looks good.